Created a starter script for fine tuning on whisper data.

arjunpatel7 opened the pull request:
I threw together a script to fine tune on @kpriyanshu256 's whisper transcripts using distilgpt2. The script is written such that we can easily pick a different model checkpoint, dataset, epochs, and block size. Could be useful for testing parts of our pipeline.

arjunpatel7 committed:
Starter script for training DistilGPT2 on our transcript data

Simple script adapted from an older huggingface script on language modeling for tensorflow. Next steps are to refactor code, maybe add command line arguments.

arjunpatel7 committed:
Cleaned up functions

Preprocessing and loading data are now wrapped in functions. Global variables for things we may want to change, such as dataset path on HF datasets, desired model checkpoint, and block size.

arjunpatel7 committed:
typo

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+    "\n",
+    "\n",
+    "#most code is taken from the old huggingface script for language modeling with \n",
+    "DATASET_PATH = \"kpriyanshu256/whisper-transcripts\"\n",
```
Can you put the parameters on another cell just ofr example to maybe use it with papermill or other library to automate the execution?

arjunpatel7 replied:
Yep, done!

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+   "source": [
+    "# create functions to read in JSON data of transcripts\n",
+    "\n",
+    "def retrieve_dataset(path):\n",
```
No need of this function you only used once to load the dataset. You can do it directly on the cell below

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+    "\n",
+    "\n",
+    "def tokenize_function(dat):\n",
+    "    tokenizer = AutoTokenizer.from_pretrained(MODEL_CHECKPOINT)\n",
```
MODEL_CHECKPOINT is a global variable as you are using a function. Can you put a variable in the definition and assign a default value? It is better if anyone wanted to change the value

arjunpatel7 replied:
This is a really good point, thank you! Change made, and will be available in  next commit.

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+    "    return tokenizer(dat[\"text\"])\n",
+    "\n",
+    "\n",
+    "def group_texts(dat):\n",
```
The same in here you have been using global variables inside a function scope. I will create a default value in the parameters of the function to be able to change it.

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+    "\n",
+    "\n",
+    "\n",
+    "def compile_model(MODEL_CHECKPOINT):\n",
```
This function you have the MODEL_CHECKPOINT as a parameter but can you change by a a lower case attribute so you can call it inside the function?
In addition, the optimizer parameters are not available if anyone wanted to tune them. Can you make them accesible?

arjunpatel7 replied:
Yes, I can surface them! For now, I will surface the ones that were used in this call. Later, we can add additional parameters as necessary.

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+   "source": [
+    "# tokenize and preprocess dataset\n",
+    "datasets = retrieve_dataset(DATASET_PATH)\n",
+    "\n",
```
Remove the extra lines.

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+    }
+   ],
+   "source": [
+    "\n",
```
Remove also here the extra lines

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+    "\n",
+    "# apply tokenization\n",
+    "tokenized_datasets = datasets.map(\n",
+    "    tokenize_function, batched=True, num_proc=4, remove_columns = [\"text\", \"id\", \"segments\"])\n",
```
Put the parameters into one line each.

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import DefaultDataCollator\n",
```
Take this import and put it at the beggining

matallanas commented on starter_training_script.ipynb:
```diff
@@ -0,0 +1,272 @@
+   ]
+  },
+  {
+   "cell_type": "code",
```
Remove this last cell

matallanas commented:
Everything seems fine but some changes can mbe made to improve the notebook.
In addition, can you:
* move the notebook to a folder called `notebooks/`?
* add some text cell explaining what you are doing to improve readability?

arjunpatel7 committed:
notebooks folder creation

arjunpatel7 committed:
moved cells for automation purposes

arjunpatel7 committed:
Fixed issues with global and local variables

I was mixing use of global and local variables inside helper functions. All functions now have default values
even when using global variables, for better use and readability.

arjunpatel7 committed:
further formatting changes

arjunpatel7 committed:
Text cell and cleanup

arjunpatel7 commented:
@matallanas, thank you for your detailed and helpful feedback! All noted changes have been made. Please let me know if further are necessary, thanks!

matallanas commented:
Everything seem perfect now :) Thanks for applying all the changes. Now it seems good.
